src/transport/raw/TCP.cpp

```diff
@@ -318,7 +318,7 @@ CHIP_ERROR TCPBase::SendAfterConnect(const PeerAddress & addr, System::PacketBuf
     return err;
 }
 
-CHIP_ERROR TCPBase::ProcessSingleMessageFromBufferHead(const PeerAddress & peerAddress, System::PacketBuffer * buffer,
+CHIP_ERROR TCPBase::ProcessSingleMessageFromBufferHead(const PeerAddress & peerAddress, System::PacketBufferHandle buffer,
```

Comment:
This one is not right.  The caller wants to retain ownership of this buffer (hence the `Retain` below before calling `HandleMessageReceived`), so we should be taking a ref, not getting ownership transferred to us.